ref/src/sw_core.F90

```diff
@@ -631,48 +549,45 @@ subroutine d2a2c_vect(sw_corner, se_corner, ne_corner, nw_corner,     &
     !----------
     ! edges:
     !----------
-    if (grid_type < 3) then
-
-      if ( js == 1 .or. jsd < npt) then
-        do j = jsd, npt-1
-          do i = isd, ied
-            utmp(i, j) = 0.5 * (u(i, j) + u(i, j+1))
-            vtmp(i, j) = 0.5 * (v(i, j) + v(i+1, j))
-          enddo
+    if ( js == 1 .or. jsd < 4) then
+      do j = jsd, 3
+        do i = isd, ied
+          utmp(i, j) = 0.5 * (u(i, j) + u(i, j+1))
+          vtmp(i, j) = 0.5 * (v(i, j) + v(i+1, j))
         enddo
-      endif
+      enddo
+    endif
 
-      if ( (je + 1) == npy .or. jed >= (npy - npt) ) then
-        do j = npy-npt+1, jed
-          do i = isd, ied
-            utmp(i, j) = 0.5 * (u(i, j) + u(i, j+1))
-            vtmp(i, j) = 0.5 * (v(i, j) + v(i+1,j) )
-          enddo
+    if ( (je + 1) == npy .or. jed >= (npy - 4) ) then
+      do j = npy-3, jed
```

Comment:
@middlecoff I could be mistaken here, but if you're replacing `npt` with 4, shouldn't this line read `do j = npy - 5, jed`

Reply:
Isn't npy-npt+1 = npy-4+1 = npy-3?

Reply:
You're right, it is. I don't know what I was thinking.